src/MultiBridgeMessageReceiver.sol

```diff
@@ -130,13 +130,15 @@ contract MultiBridgeMessageReceiver is IMultiBridgeMessageReceiver, ExecutorAwar
         /// increment quorum
         ++msgDeliveryCount[msgId];
 
-        /// stores the execution data required
-        ExecutionData memory prevStored = msgExecData[msgId];
+        /// stores the hash of the execution data required
+        bytes32 prevStoredHash = msgExecDataHash[msgId];
 
         /// stores the message if the amb is the first one delivering the message
-        if (prevStored.target == address(0)) {
-            msgExecData[msgId] = ExecutionData(
-                _message.target, _message.callData, _message.nativeValue, _message.nonce, _message.expiration
+        if (prevStoredHash == bytes32(0)) {
+            msgExecDataHash[msgId] = keccak256(
```

Comment:
I think it might be cleaner if we refactor a bit as follows:
- Rename `ExecutionData` to `MessageExecutionParams` 
- Move `MessageExecutionParams` to the library `MessageLibrary.sol`
- Add utility function for extracting `MessageExecutionParams` from a `Message` in `MessageLibrary`
- Add utility functions for computing execution hash, given a `Message` or a `MessageExecutionParams`
- use these utility functions in this contract and other places as needed

Reply:
Done.